core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/FileSink.java

```diff
@@ -0,0 +1,282 @@
+      recordTransformer = input -> {
+        try {
+          GenericRecord record = avroTransformer.transform(input);
+          return new KeyValue<>(null, record);
+        } catch (IOException e) {
+          throw new RuntimeException(e);
+        }
+      };
+    } else if ("text".equalsIgnoreCase(config.format)) {
+      recordTransformer = input -> {
+        List<String> dataArray = new ArrayList<>();
+        for (Schema.Field field : input.getSchema().getFields()) {
+          Object fieldValue = input.get(field.getName());
+          String data = (fieldValue != null) ? fieldValue.toString() : NULL_STRING;
+          dataArray.add(data);
+        }
+        return new KeyValue<>(new Text(Joiner.on(config.delimiter).join(dataArray)), NullWritable.get());
+      };
+    }
+  }
+
+  @Override
+  public void transform(StructuredRecord input, Emitter<KeyValue<Object, Object>> emitter) throws Exception {
+    emitter.emit(recordTransformer.apply(input));
+  }
+
+  /**
+   * Config for File Sink.
+   */
+  public static class Conf extends ReferencePluginConfig {
+    private static final Gson GSON = new Gson();
+    private static final Type MAP_TYPE = new TypeToken<Map<String, String>>() { }.getType();
+    private static final String TEXT = "text";
+    private static final String AVRO = "avro";
+    private static final String PARQUET = "parquet";
+
+    @Macro
+    @Description("Destination path prefix. For example, 'hdfs://mycluster.net:8020/output'")
+    private String path;
+
+    @Macro
+    @Description("Time suffix used for the destination directory for each run. For example, 'YYYY-MM-dd-HH-mm'. " +
+      "By default, no time suffix is used.")
+    @Nullable
+    private String suffix;
+
+    @Macro
+    @Nullable
+    @Description("Advanced feature to specify any additional properties that should be used with the sink.")
+    private String fileSystemProperties;
+
+    @Macro
+    @Nullable
+    @Description("The delimiter to use when concatenating record fields and the format is 'text'. "
+      + "Defaults to a comma (',').")
+    private String delimiter;
+
+    @Macro
+    @Nullable
+    @Description("The format to write in. Must be 'text', 'avro', or 'parquet'. Defaults to 'text'.")
+    public String format;
+
+    @Macro
+    @Nullable
+    @Description("The schema to use when the format is 'avro' or 'parquet'. If none is specified, the input schema "
+      + "will be used.")
+    public String schema;
+
+    private Conf() {
+      super("");
+      fileSystemProperties = "{}";
+      delimiter = ",";
+      format = "text";
+    }
+
+    private String getOutputDir(long logicalStartTime) {
+      String timeSuffix = !Strings.isNullOrEmpty(suffix) ? new SimpleDateFormat(suffix).format(logicalStartTime) : "";
+      return String.format("%s/%s", path, timeSuffix);
+    }
+
+    @Nullable
+    private Schema getSchema(@Nullable Schema inputSchema) {
+      if (containsMacro("schema")) {
```

Comment:
is this check required ? getSchema is called only if its not a macro in validate.

Reply:
it's not currently called, but thought it would be better to handle the scenario in this method too since it's pretty tricky.

Reply:
ok 👍